### ISSUE TITLE
[BUG] - Reflect panic with too many jobs possibly 

### ISSUE THREAD

**Issue by JohnRoesler** [go-co-op/gocron#84]
### Describe the bug

reported by @Algalish

```
panic: reflect: call of reflect.Value.Type on zero Value

goroutine 44934 [running]:
reflect.Value.Type(0x0, 0x0, 0x0, 0x0, 0x175fdd5aee8)
        /usr/local/go/src/reflect/value.go:1877 +0x166
github.com/go-co-op/gocron.callJobFuncWithParams(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x175fe0375a8, 0x40fd526000000000, 0x0, 0x0)
       /projects/src/src/github.com/go-co-op/gocron/gocron.go:53 +0xc5
github.com/go-co-op/gocron.(*Job).run(0xc0000dc540)
        /projects/src/src/github.com/go-co-op/gocron/job.go:52 +0x13b
created by github.com/go-co-op/gocron.(*Scheduler).run
        /projects/src/src/github.com/go-co-op/gocron/scheduler.go:290 +0x3f
```

### To Reproduce

```
cronScheduler = gocron.NewScheduler(time.UTC)
	for h := 0; h < 24; h++ {
		for m := 0; m < 60; m += 20 {
			timeString := fmt.Sprint(h) + ":" + fmt.Sprint(m)
			cronScheduler.Every(1).Day().At(timeString).Do(getModDetails)
			log.Println(timeString)
		}
	}
```

**Comment by arjunmahishi** [comment]
I modified the `for` loop a little bit to debug this:
```go
for h := 0; h < 24; h++ {
	for m := 0; m < 60; m += 20 {
		timeString := fmt.Sprint(h) + ":" + fmt.Sprint(m)
		fmt.Println("timeString: ", timeString)
		if _, err := cronScheduler.Every(1).Day().At(timeString).Do(task); err != nil {
			panic(err)
		}
		log.Println(timeString)
	}
}
```
This is the output I got:

```text
timeString:  0:0
panic: time format error

goroutine 1 [running]:
main.main()
        /tmp/test/main.go:22 +0x5f9
exit status 2
```

As you can see, there is an error while running the `Do` function. But since the error is not being handled, the `panic` happens. 

This is happening because we create a placeholder job in the `Every` function. And we only assign a "task"/"function" to that job in the `Do` function. But in this case, we are returning early in the `Do` function, leaving the `job.funcs` to be empty. Later when we call  `callJobFuncWithParams`, the `jobFunc` passed to it is `nil`. And doing a `.Type()` on this `nil` is what is causing the above panic. 

The simple fix for this would be to remove this placeholder job from the scheduler while returning any error from the `Do` function. Making a PR in sometime. If there is a better way to handle this, we can discuss it in the PR.

**Comment by JohnRoesler** [comment]
Resolved in [v0.4.0](https://github.com/go-co-op/gocron/releases/tag/v0.4.0)
